similar/main/songs.cpp

```diff
@@ -125,6 +150,26 @@ static void songs_init()
 		{
 			if ( strlen( inputline ) )
 			{
+				if (canUseExtensions)
+				{
+					// extension stuffs
+					if (!d_strnicmp(inputline, "!Rebirth.secret ", 16)) {
+						BIMSecretSongs = reinterpret_cast<bim_song_info *>(d_realloc(BIMSecretSongs, sizeof(bim_song_info) * (Num_secret_bim_songs + 1)));
+						memset(BIMSecretSongs[Num_secret_bim_songs].filename, '\0', sizeof(BIMSecretSongs[Num_secret_bim_songs].filename));
+						sscanf( inputline + 16, "%15s", BIMSecretSongs[Num_secret_bim_songs].filename );
```

Comment:
**might**, but 8.3 is 8.3 and the assumption is that whitespace isn't supported, since it isn't on DOS either. To my knowledge, nobody has tried to put whitespace on a filename, but it's something that could be changed regardless.

Reply:
This doesn't use the 8.3 rules though.  It allocates 15 bytes + null terminator for the filename.